akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala

```diff
@@ -28,10 +28,10 @@ import scala.annotation.switch
   final val Tab = '\t'.toByte
   final val Space = ' '.toByte
 
-  final val Whitespace = Set(LineBreak, LineBreak2, Tab, Space)
-
-  def isWhitespace(input: Byte): Boolean =
-    Whitespace.contains(input)
+  val isWhitespace: (Byte) ⇒ Boolean = {
+    case LineBreak | LineBreak2 | Tab | Space ⇒ true
+    case _                                    ⇒ false
+  }
```

Comment:
I do think going over plain method over val of Function.apply is quicker, seem to confirm it in benchmarks as well; 
Such version:

```
```

is `[info] JsonFramingBenchmark.counting_long_document  thrpt    3  203.913 ± 28.216  ops/s`

while the one from PR is `[info] JsonFramingBenchmark.counting_long_document  thrpt    3  172.621 ± 36.822  ops/s`

Reply:
And by making it table switch we gain even more:

```scala
// compile time constants to allow a table-switch isWhitespace
  final val LineBreak = 10 // '\n'
  final val LineBreak2 = 13 // '\r'
  final val Tab = 9 // '\t'
  final val Space = 32 // ' '

  def isWhitespace(b: Byte): Boolean =
    (b: @switch) match {
      case Space      ⇒ true
      case LineBreak  ⇒ true
      case LineBreak2 ⇒ true
      case Tab        ⇒ true
      case _          ⇒ false
    }
```
sometimes even gets to 210-ish range... 

`[info] JsonFramingBenchmark.counting_long_document  thrpt    5  203.836 ± 10.730  ops/s`

```
[info] # Fork: 1 of 1
[info] # Warmup Iteration   1: 236.365 ops/s
[info] # Warmup Iteration   2: 240.669 ops/s
[info] # Warmup Iteration   3: 213.602 ops/s
[info] Iteration   1: 204.050 ops/s
[info] Iteration   2: 206.229 ops/s
[info] Iteration   3: 201.337 ops/s
[info] Iteration   4: 200.696 ops/s
[info] Iteration   5: 206.866 ops/s
```

Reply:
If you'd like to include the table switch one please do @lustefaniak :)
Makes sense to optimize this bit, gets called pretty often.

Reply:
Interesting, I tried that with both, def+match and with table switch and they were slower than function val. I just checked, that I was using GraalVM for those benchmarks I run, so maybe this is the source of the difference. 
I will now repeat benchmarks with ```java version "1.8.0_172"
Java(TM) SE Runtime Environment (build 1.8.0_172-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.172-b11, mixed mode)```

Reply:
Hah yeah, we're not targeting graal as the go-to JVM people are using :)
Give it a spin on JDK8 or 9 and let's merge what's nice there. I think Graal probably was able to inline the Function.apply nicely and plain old JIT did not figure that out :)

Reply:
So either benchmarks are flawed on GraalVM or it really does amazing job optimizing that code 🙀 

I checked it on HotSpot, and it is much better to use table switch, `~124 -> ~210`. On the GraalVM I don't see such large difference, but base performance is much better anyway `~337 -> ~370` ( I have some app in the background which I can't stop now so the numbers are little bit lower than in my comments)

Reply:
Nice :-) Yes, Graal JIT is known to perform very well on Scala code. Nice that the table switch helped it so much as well!